src/containers/methodCallContainer.js

```diff
@@ -1,55 +1,134 @@
 import { navigate, useParams } from '@reach/router';
 
+const CONTRACT_FUNCTION_METHOD = 'eth_call';
+
 const MethodCallContainer = () => {
   const params = useParams();
-  const { codeSampleVisible, toggleSampleCode } = useContext(AppContext);
+  const { codeSampleVisible, toggleSampleCode, abi, setAbi } = useContext(
+    AppContext
+  );
   const { addToLog } = useContext(LogContext);
   const logItem = useCallback(addToLog, []);
+
   const {
     web3URL = '',
     web3Lib = '',
     currentMethod = '',
     formArgs = '',
   } = params;
-  const argumentList = formArgs.split('/');
+
   const web3Method = Web3RpcCalls[currentMethod] || {};
   const { description, disabled } = web3Method || {};
-  const { args, exec } = web3Method[web3Lib] || {};
-  const setArgumentList = (val, index) => {
-    const argsCopy = [...argumentList];
-    argsCopy[index] = val;
-    // bad pattern?
-    let joinedArgs = argsCopy.join('/');
+  const { args: initialFormInputs, exec } = web3Method[web3Lib] || {};
+
+  const [formInputs, setFormInputs] = useState([]);
+  const [argumentList, setArgumentList] = useState([]);
+
+  const updateURL = (val, index) => {
+    let argsList = formArgs.split('/').slice(0, formInputs.length); // Remove dangling arguments
+    argsList[index] = val;
+    let joinedArgs = argsList.join('/');
     let url = `/${web3URL}/${web3Lib}/`;
     if (currentMethod) url += `${currentMethod}/`;
-    if (args.length > 0) url += `${joinedArgs}`;
+    if (formInputs.length > 0) url += `${joinedArgs}`;
     navigate(url);
   };
-  const runRequest = (args) => {
+
+  const onUpdateArguments = async (val, index) => {
+    if (currentMethod === CONTRACT_FUNCTION_METHOD) {
+      if (index === 1) {
+        // Prevent updating URL if ABI error
+        const { error } = await fetchOrParseAbi(val);
+        if (error)
+          return logItem({
+            method: 'error',
+            data: ['🚨 Error:', error],
+          });
+        return updateURL(btoa(val), index);
+      }
+    }
+    updateURL(val, index);
+  };
+
+  const runRequest = () => {
     logItem({
       method: 'info',
       data: [`🚀 Sending request for **${currentMethod}**:`],
     });
     const [provider, proto] = buildProvider(web3Lib, atob(web3URL));
-    exec(provider, proto, ...argumentList)
-      .then((response) => {
+    let args = argumentList.slice();
+    // Pre-flight conversion for contract calls
+    if (currentMethod === CONTRACT_FUNCTION_METHOD)
+      args = getContractFriendlyArguments(args, abi);
+    try {
+      exec(provider, proto, ...args).then((response) => {
         logItem({
           method: 'info',
           data: [`✅ Node response:`, response],
         });
-      })
-      .catch((err) => {
-        logItem({
-          method: 'error',
-          data: ['🚨 Error response:', err],
-        });
       });
+    } catch (err) {
+      logItem({
+        method: 'error',
+        data: ['🚨 Error response:', err],
+      });
+    }
```

Comment:
Why the refactor from `.catch` to `try/catch`? I believe the latter is used for `async/await`.

Reply:
Was debugging and this is just what I'm used to. I changed it back to .catch()